src/vmemcache.c

```diff
@@ -120,7 +120,7 @@ vmemcache_set_size(VMEMcache *cache, size_t size)
 		return -1;
 	}
 
-	if (size >= (1ULL << 56)) {
+	if (size >= ((sizeof(void *) > 4) ? (1ULL << 56) : -1UL)) {
```

Comment:
Please don't rely on implicit casting.

This will trigger only for one value, so I wonder whether this check is needed at all on 32-bit....